### DISCUSSION TITLE
Interested in translations?

### DISCUSSION THREAD

**Discussion by renewoensdregt** [DurgNomis-drol/ha_toyota#41] [Ideas]
Hi, 
As I am not a good coder, but glad to be willing to help out, I am happy to help out in translating as I see currently we have Danish and English, but I could help you with Dutch, Spanish, Catalan and Swedish.
if you let me know how I can do that, happy to help!
R

**Comment by DurgNomis-drol** [comment]
They are more then welcome!! If you need any help just ping me here 😃

**Reply by renewoensdregt** [reply]
I have seen some language.json files, but I am not sure how to proceed? Would that just be creating an es.json file for example and dropping that one here with the translation? or do I do that in some other way?

**Reply by DurgNomis-drol** [reply]
Yes and then just open a PR with them so i can add them to the next release 😄

**Reply by renewoensdregt** [reply]
ok.. I´ll give that a shot! Even working in Git is new to me haha! learning every day!

**Reply by renewoensdregt** [reply]
Should I create one PR for each language or one for the couple I can do?
I am also confused about the language code. Why would we need to do it in the form of da-dk when region is not used? should we not refer to a table like this https://www.loc.gov/standards/iso639-2/php/code_list.php and only use the language code?

**Reply by DurgNomis-drol** [reply]
Yes that is preferable 😃 

The locale code you are referring to does not have anything to do with the language selection. HA handles this internally and we (Custom integration maintainers) just need to provide a translation file with the language code.

The locale code is used by the integration to connect to Toyota's servers.

**Reply by renewoensdregt** [reply]
OK will do one by one. 
How do I know what the right code is for each language? I have to say that that confused me even in the current version. I always enter da-dk, even if my system is in English as I could not get it to work and that does not to impact anything.

**Reply by DurgNomis-drol** [reply]
The locale code you enter when you setup the integration does not effect the language at all. It is used internally to connect you to Toyota's servers. 

If the language you have chosen for your Home Assistant installation is not in the translation folder (Example `/translations/da.json`), it will default to English. So that is is why it didn't work for you.

The language code format you can find [here](https://developers.home-assistant.io/docs/internationalization/custom_integration)

**Reply by renewoensdregt** [reply]
I really have no clue if I did it all right...
To add the translation file and the PR, I first seem to have had to fork the repository). I chose the master.. not sure if that is right), add the file and then added the file. I am quite sure somewhere in the code you´d have to choose the right file based on the new language.

let me know if what I did looks right before I add the others in the same way.

**Reply by renewoensdregt** [reply]
I looked in this link "The language code format you can find [here](https://developers.home-assistant.io/docs/internationalization/custom_integration)" and not sure how you found the ca for catalan.... don´t see a list of language codes there...

**Reply by DurgNomis-drol** [reply]
I Google'd a bit more around and found out that it follows this standard ISO 639-1.  😃

**Reply by renewoensdregt** [reply]
Added the Dutch language now too. hope you can see it this time (PR 48). Deleted my old repository I cloned and created a new one. hope that is the way to do it.

**Reply by DurgNomis-drol** [reply]
Yes, and it is merged! You did it correctly 😃

**Reply by renewoensdregt** [reply]
Perfect! Now that you have merged it I can delete my repository again, right? In case of new PRs, then I can fork a new repository, right?

**Reply by DurgNomis-drol** [reply]
Yes you can 😃
